homeassistant/components/template/event.py

```diff
@@ -0,0 +1,270 @@
+            vol.Optional(CONF_PICTURE): cv.template,
+            vol.Optional(CONF_UNIQUE_ID): cv.string,
+        }
+    )
+    .extend(TEMPLATE_ENTITY_ATTRIBUTES_SCHEMA.schema)
+    .extend(TEMPLATE_ENTITY_AVAILABILITY_SCHEMA.schema)
+    .extend(TEMPLATE_ENTITY_ICON_SCHEMA.schema)
+)
+
+EVENT_CONFIG_SCHEMA = vol.Schema(
+    {
+        vol.Optional(CONF_DEVICE_CLASS): DEVICE_CLASS_SCHEMA,
+        vol.Optional(CONF_DEVICE_ID): selector.DeviceSelector(),
+        vol.Required(CONF_EVENT_TYPE): cv.template,
+        vol.Required(CONF_EVENT_TYPES): cv.template,
+        vol.Required(CONF_NAME): cv.template,
+    }
+)
+
+
+async def _async_create_entities(
+    hass: HomeAssistant, definitions: list[dict[str, Any]], unique_id_prefix: str | None
+) -> list[TemplateEvent]:
+    """Create the Template event."""
+    entities = []
+    for definition in definitions:
+        unique_id = definition.get(CONF_UNIQUE_ID)
+        if unique_id and unique_id_prefix:
+            unique_id = f"{unique_id_prefix}-{unique_id}"
+        entities.append(TemplateEvent(hass, definition, unique_id))
+    return entities
+
+
+async def async_setup_platform(
+    hass: HomeAssistant,
+    config: ConfigType,
+    async_add_entities: AddEntitiesCallback,
+    discovery_info: DiscoveryInfoType | None = None,
+) -> None:
+    """Set up the template select."""
+    if discovery_info is None:
+        _LOGGER.warning(
+            "Template event entities can only be configured under template:"
+        )
+        return
+
+    if "coordinator" in discovery_info:
+        async_add_entities(
+            TriggerEventEntity(hass, discovery_info["coordinator"], config)
+            for config in discovery_info["entities"]
+        )
+        return
+
+    async_add_entities(
+        await _async_create_entities(
+            hass, discovery_info["entities"], discovery_info["unique_id"]
+        )
+    )
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Initialize config entry."""
+    _options = dict(config_entry.options)
+    _options.pop("template_type")
+    validated_config = EVENT_CONFIG_SCHEMA(_options)
+    async_add_entities([TemplateEvent(hass, validated_config, config_entry.entry_id)])
+
+
+class AbstractTemplateEvent(EventEntity):
+    """Representation of a template event features."""
+
+    def __init__(self, config: dict[str, Any]) -> None:
+        """Initialize the features."""
+        self._event_type_template = config[CONF_EVENT_TYPE]
+        self._event_types_template = config[CONF_EVENT_TYPES]
+
+        self._attr_device_class = config.get(CONF_DEVICE_CLASS)
+
+        self._event_type = None
+        self._attr_event_types = []
+
+    @property
+    def event_types(self) -> list[str]:
+        """Return the event_types list."""
+        return self._attr_event_types
```

Comment:
```suggestion
```
This is the default implementation

Reply:
If I remove that, the template doesn't update the value because it's a cached_property on the base implementation.  Using @property allows the template to dynamically update the list when the template changes.

Reply:
but in other integrations we can also freely use `_attr_event_types` and other shorthands without caching issues

Reply:
I can try again, but for some reason it isn't pulling the updated lists into the state machine.